articles/151_node_to_participant_mapping.md

```diff
@@ -0,0 +1,213 @@
+
+A message could be send for:
+- Each `Node`
+- Each `Participant`
+
+The second option reduces the amount of messages and it does lifetime management easier, as the message will not be more available after the `Participant` is deleted (using the correct QoS policies).
+In the following, the second option will be considered.
+
+##### State Message
+
+Each `Participant` will send a message representing their state.
+A keyed topic could be used for communicating it.
+It's reasonable to organize the data by `Participant`, so the `Participant` guid can be used as the key.
+The rest of the message will be a vector of with the information of each node.
+That message should contain the `Node` name, and four vectors:
+- guid of its `Publishers`
+- guid of its `Subscriptions`
+- guid of its `Services`
+- guid of its `Clients`
+
+Vector bounds: TBD
+
+##### QoS for communicating node information
+
+Each published message should be available to late subscriptions, and only the last message of each key is needed.
+For that reason, the QoS of the `Publishers` should be:
+
+- Durability: Transient Local
+- History: Keep Last
+- History depth: 1
+- Reliability: Reliable
+
+Considering that a keyed topic will be used, in which the history depth apply for each key, only one `Publisher` per process will be needed.
+
+The configuration of the `Subscriber` QoS depends on how the data will be accessed later:
+- Polled using `Subscriber` read method when needed.
+- Listened and organized in a local cache.
+
+The second option allows better organization of this information (e.g.: in hash tables).
+In the first case, the QoS of the `Subscriber` should be:
+
+- Durability: Transient Local
+- History: Keep Last
+- History depth: 1
+- Reliability: Reliable
+
+In the second case, durability can be changed to `Volatile`.
+
+#### Using USER_DATA and GROUP_DATA QoSPolicy
+
+Each `Participant` could have in its user data the list of node names that owns.
+When this data is changed, each `ParticipantListener` will be notified.
+This is not a good option, as `UserData` is just a sequence of bytes, and communicating a list of node names in it doesn't seem as a reasonable solution.
+
+Similarly to `UserData`, `GroupData` is a available in `Publishers` and `Subscribers`.
+These entities only need to communicate the guid of the `Participant` and the `Node` name from which it was created.
+This idea can be combined with a topic just publishing the list of `Node` names, without including all the other vectors in the message.
+Although, it is more difficult to communicate this information for `Services` and `Clients`, as they use behind the scenes just a `DDS Publisher` and `Subscriber`.
+
+### Other implications
+
+#### Security
+
+In `DDS`, security can be specified at a `Participant` level.
+Currently, as each `Node` was mapped to one `Participant`, we can individually configure its security key and access control policy.
```

Comment:
@ivanpauno I'd avoid saying `currently` and rephrase so that this article remains valid in the future.